rmw_connext_shared_cpp/src/shared_functions.cpp

```diff
@@ -319,6 +323,10 @@ create_node(const char * implementation_identifier, const char * name, size_t do
   }
   participant_qos.participant_name.name =
     static_cast<char *>(rmw_allocate((strlen(name) + 1) * sizeof(char)));
+  if (!participant_qos.participant_name.name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory");
+    return NULL;
+  }
   participant_qos.participant_name.name = strdup(name);
```

Comment:
See 27ffa13 for my guess at a fix, I'm still waiting for it to rebuild so I can check for double free's.